lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1410,15 +1411,27 @@ sub failed_modules ($self) {
 }
 
 sub update_status ($self, $status) {
+    my $jobs = $self->result_source->schema->resultset('Jobs');
     my $ret = {result => 1};
 
-    # that is a bit of an abuse as we don't have anything of the
-    # other payload
+    # set job to UPLOADING if it is still executed
+    # note: That is a bit of an abuse as we don't have anything of the
+    #       other payload.
     if ($status->{uploading}) {
-        $self->update({state => UPLOADING});
-        return $ret;
+        my %cond = (id => $self->id, state => {-in => [EXECUTION_STATES]});
+        return {result => $jobs->search(\%cond)->update({state => UPLOADING}) != 0};
     }
 
+    # set job to RUNNING if it is still ASSIGNED/SETUP
+    my %cond = (id => $self->id, state => {-in => [ASSIGNED, SETUP]});
+    $jobs->search(\%cond)->update({state => RUNNING, t_started => now()});
+
+    # abort any further updates when we couldn't set the job to RUNNING
+    # note: That can be the case if the concurrently running stale job detection
+    #       wins the race updating the job state.
+    $self->discard_changes;
+    return {result => 0} unless $self->state eq RUNNING;
```

Comment:
I don't see anything here that makes use of the result. Previously it was always 1 so it might be good to have a unit test that depends on the difference.